docs/en/cookbook/generated-columns.rst

```diff
@@ -0,0 +1,99 @@
+Generated Columns
+=================
+
+Generated columns, sometimes also called virtual columns, are populated by
+the database engine itself. They are a tool for performance optimization, to
+avoid calculating a value on each query.
+
+You can define generated columns on entities and have doctrine map the values
+to your entity.
+
+Declaring a generated column
+----------------------------
+
+There is no explicit mapping instruction for generated columns. Instead, you
+specify that the column should not be written to, and define a custom column
+definition.
+
+.. code-block:: php
+
+    <?php
+
+    use Doctrine\ORM\Mapping as ORM;
+
+    #[ORM\Entity]
+    class Person
+    {
+        #[ORM\Column(type: "string")]
+        private $firstName;
+
+        #[ORM\Column(type: "string")]
+        private $lastName;
+
+        #[ORM\Column(
+            type: "string",
+            insertable: false,
+            updatable: false,
+            columnDefinition: "VARCHAR(255) GENERATED ALWAYS AS (concat(first_name, ' ', last_name) stored NOT NULL",
+            generated: "ALWAYS",
+        )]
+        private $fullName;
+    }
```

Comment:
Now that we are using `phpDocumentor/guides`, it's possible to use `literalinclude`. Here is an example:

https://github.com/doctrine/orm/blob/f8fdeaf41eec0aa8882c7d62aa31ea4732b0ce7b/docs/en/tutorials/working-with-indexed-associations.rst?plain=1#L32-L36

I haven't dared migrate existing docs to that because I think it would make the merge ups difficult, but I think for new code it's fine.

Reply:
yeah i agree that migrating them is tricky. but for new docs its indeed quite neat - my IDE then also helps me write the code cleanly.